aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -82,16 +90,19 @@ private AWSDataStorePlugin(
             @NonNull ModelProvider modelProvider,
             @NonNull ModelSchemaRegistry modelSchemaRegistry,
             @NonNull ApiCategory api,
-            @Nullable DataStoreConfiguration userProvidedConfiguration) {
+            @Nullable DataStoreConfiguration userProvidedConfiguration,
+            @NonNull Reachability reachability) {
         this.sqliteStorageAdapter = SQLiteStorageAdapter.forModels(modelSchemaRegistry, modelProvider);
         this.categoryInitializationsPending = new CountDownLatch(1);
         this.api = api;
+        // A host to check periodically, for an internet connection
```

Comment:
What if we used the /ping endpoint of the AWS AppSync control plane? Not exactly right, but I think it's better than the website.

```console
$ curl -s -D -  https://appsync.us-east-1.amazonaws.com/ping 

HTTP/2 200 
content-type: null
content-length: 7
date: Thu, 21 May 2020 23:01:54 GMT
x-cache: Miss from cloudfront
via: 1.1 5195de19cbc5ce842ac6538e9a6850cb.cloudfront.net (CloudFront)
x-amz-cf-pop: IAD89-C1
x-amz-cf-id: 8U67lKT86LfcFyoPZO0uuQ9uLgjLI9F_Lizo5qoJNaauWf9v5YY29w==

healthy
```

Reply:
Another idea is to just not do this at all, and just test if we're connected to a network (not necessarily the internet) and backoff and retry as errors come.

Reply:
What do iOS and JS do?

Reply:
I'm not sure about iOS, but if it's like JS the current implementation does not check an endpoint for reachability however this is something that Francisco would like to do we just haven't had the time. If we do add it this should be using the realtime WSS endpoint with a ping. First, just hitting any old URL doesn't guarantee the service endpoint is reachable nor does the failure of that imply the opposite. Customers will be using clients in a myriad of situations including VPNs with location whitelists or even random DNS problems impacting one location rather than another. Additionally AWS services are architected to maximize availability of the Data Plane and not the Control Plane, so the CP going down shouldn't impact the client's ability to perform normally against the DP.

My recommendation is if we have this functionality to ping as an additional variable for network detection to use the AppSync Endpoint from config for either HTTP or WSS (or a combo of both). If not enough time to do that for GA we should remove this for initial release as it's an optimization.

Reply:
I think the Control Plane endpoint is a bad idea given it isn’t a contract, and I agree it isn’t testing what we want. I think we should punt, but change this check to see if we’re connected to a network.